### DISCUSSION TITLE
Feature Request: Actionable Tasks for Atomic Calendar

### DISCUSSION THREAD

**Discussion by apzuckerman** [totaldebug/atomic-calendar-revive#1598] [Feature Requests]
Feature Request: Actionable Tasks for Atomic Calendar

**Feature Description:**

It would be incredibly useful to add functionality to Atomic Calendar that allows users to mark calendar events as actionable tasks. When a task is marked as actionable, it could be cleared (or completed) by unchecking it directly in the card. This would enable more dynamic task management within the calendar view.

**Additionally, a dedicated card could display:**

1. Upcoming actionable tasks (within a user-defined number of days, e.g., the next 7 days).
2. Uncleared tasks (persisting tasks that haven't been marked complete, spanning from today backward).

**Why This Feature Is Important:**

1. Enhances Atomic Calendar’s utility by bridging calendar events and actionable task management.
2. Provides users with a simple way to manage recurring or event-based tasks directly in the calendar interface.
3. Improves task tracking by showing overdue or uncompleted items alongside upcoming priorities.

**Proposed Implementation:**

1. Add an option to mark a calendar event as an actionable task.
2. Include a visual indicator (e.g., a checkbox) for actionable tasks within calendar views.
3. Create a new card type (or extend the existing card) to filter:
4. Upcoming actionable tasks: Tasks due within a specified future range.
5. Uncleared tasks: Tasks not yet completed from past days.

Would this be possible to implement? Happy to provide additional feedback or testing if needed!

**Comment by marksie1988** [comment]
This goes beyond the scope of the calendar card, it's unlikely I will spend time implementing something like this.

**Comment by apzuckerman** [comment]
Sounds good, thanks for the reply. Seems like there are a bunch of posts from people trying to figure out a solution (generally). If you ever change your mind, I'd love to test.

**Reply by marksie1988** [reply]
Unfortunately I don't have the time to work on something like that as it would be a decent piece work to complete. 

If someone else came up with a plan and design etc I would consider it and then they could implement it but I started a new job a few months ago and I don't have much time anymore.
